### PR TITLE
cephfs-journal-tool: add "set pool_id" option

### PR DESCRIPTION
**Pull request by taodd**
1 Fixes issue: http://tracker.ceph.com/issues/22631
2 add error log info to give user a hint
3 complement the usage info for header set <field>

### CODE REVIEW

**Comment by taodd** [comment]
retest this please

File: src/osdc/Journaler.cc [hunk 1]
```diff
@@ -84,7 +84,11 @@ void Journaler::_set_layout(file_layout_t const *l)
 {
   layout = *l;
 
-  assert(layout.pool_id == pg_pool);
+  if (layout.pool_id != pg_pool) {
+    // user can reset pool id through cephfs-journal-tool
+    lderr(cct) << "may got older pool id from header layout" << dendl;
+    assert(0);
```
Comment (batrick) [inline src/osdc/Journaler.cc]: Use `ceph_abort()`.

Comment (taodd) [reply src/osdc/Journaler.cc]: @batrick done

**Comment by ukernel** [comment]
LGTM

**Pull request merged**